modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -233,7 +248,7 @@ object JacksonGenerator {
                 new BlockStmt(
                   new NodeList(
                     new IfStmt(
-                      new MethodCallExpr("value.name.equals", new NameExpr("name")),
+                      new MethodCallExpr("value.name.toString().equals", new NameExpr("name")),
```

Comment:
I would really much prefer something like
```scala
tpe match {
  case pt: PrimitiveType => new BinaryExpr(/* use '==' to check */)
  case other => // use toString
}
```

Reply:
I was just trying to preserve the `fromString` function type, but the more I think about it, we really should be `MyEnum.fromValue(5l): Optional<MyEnum>` or something, having the methods be hard-coded to `String` seems like a mistake

Reply:
Unfortunately we have to provide `fromString()` as that's a magic method name that Jersey will look for if you use an enum type as a resource method parameter.

Reply:
(I'm cool with adding a safer version for our own use, though.)

Reply:
So, after doing some offline conversation, I think leaving this the way it is actually makes the most sense, at least for now. `.toString()` should be a no-op on `String`, and other types will need to be looked up with this magic function via Jersey, so we can't use `==` as you indicate in the original comment.

Please feel free to correct me if I misunderstood, but I'll consider this resolved for now.